web-server/src/content/Dashboards/ConfigureGithubModalBody.tsx

```diff
@@ -0,0 +1,275 @@
+                e.stopPropagation();
+                e.nativeEvent.stopImmediatePropagation();
+                handleSubmission();
+                return;
+              }
+            }}
+            error={!!showError.value}
+            sx={{ width: '100%' }}
+            value={token.value}
+            onChange={(e) => {
+              handleChange(e.currentTarget.value);
+            }}
+            label="Github Personal Access Token"
+            type="password"
+          />
+          <Line error tiny mt={1}>
+            {showError.value}
+          </Line>
+          <FlexBox>
+            <Line tiny mt={1} primary sx={{ cursor: 'pointer' }}>
+              <Link
+                href="https://github.com/settings/tokens"
+                target="_blank"
+                rel="noopener noreferrer"
+              >
+                <Line
+                  underline
+                  sx={{
+                    textUnderlineOffset: '2px'
+                  }}
+                >
+                  Generate new classic token
+                </Line>
+              </Link>
+              <Line ml={'5px'}>{' ->'}</Line>
+            </Line>
+          </FlexBox>
+        </FlexBox>
+
+        <FlexBox justifyBetween alignCenter mt={'auto'}>
+          <FlexBox col sx={{ opacity: 0.8 }}>
+            <Line>Learn more about Github</Line>
+            <Line>
+              Personal Access Token (PAT)
+              <Link
+                ml={1 / 2}
+                href="https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens"
+                target="_blank"
+                rel="noopener noreferrer"
+              >
+                here
+              </Link>
+            </Line>
+          </FlexBox>
+          <FlexBox gap={2} justifyEnd>
+            <LoadingButton
+              loading={isLoading.value}
+              variant="contained"
+              onClick={handleSubmission}
+            >
+              Confirm
+            </LoadingButton>
+          </FlexBox>
+        </FlexBox>
+      </FlexBox>
+      <Divider orientation="vertical" flexItem />
+      <TokenPermissions />
+    </FlexBox>
+  );
+};
+
+const TokenPermissions = () => {
+  const imageLoaded = useBoolState(false);
+
+  const expandedStyles = useMemo(() => {
+    const baseStyles = {
+      border: `2px solid ${alpha('rgb(256,0,0)', 0.4)}`,
+      transition: 'all 0.8s ease',
+      borderRadius: '12px',
+      opacity: 1,
+      width: '240px',
+      position: 'absolute',
+      maxWidth: 'calc(100% - 48px)',
+      left: '24px'
+    };
+
+    return [
+      {
+        height: '170px',
+        top: '58px'
+      },
+      {
+        height: '42px',
+        top: '230px'
+      },
+      {
+        height: '120px',
+
+        top: '378px'
+      },
+      {
+        height: '120px',
+
+        top: '806px'
+      }
+    ].map((item) => ({ ...item, ...baseStyles }));
+  }, []);
+
```

Comment:
seems repeated

Reply:
Naah, base styles are slightly different, refactor would prolly just make it more messy.